lib/s3middleware/nullStream.js

```diff
@@ -0,0 +1,42 @@
+const Readable = require('stream').Readable;
+
+
+/**
+ * This class is used to produce zeros filled buffers for a reader consumption
+ */
+class NullStream extends Readable {
+
+    /**
+     * Construct a new zeros filled buffers producer that will
+     * produce as much bytes as specified by the range parameter, or the size
+     * parameter if range is null or not constituted of 2 elements
+     * @constructor
+     * @param {integer} size - the number of null bytes to produce
+     * @param {array} range - a range specification to override to size
+     */
+    constructor(size, range) {
```

Comment:
IMO, this class should not have to deal with ranges, with the goal of making this API as simple as it could/should be for a generic implementation I would recommend to only support the `size` parameter and let the user do his range to size conversion on his own if he has to.

Reply:
Makes sense. But in this particular case, it mimics the API of the data wrapper client ```get``` method.